server/mergin/sync/public_api_controller.py

```diff
@@ -813,7 +816,9 @@ def project_push(namespace, project_name):
         if not all(ele.path != item.path for ele in project.files):
             abort(400, f"File {item.path} has been already uploaded")
         if not is_valid_path(item.path):
-            abort(400, f"File {item.path} contains invalid characters.")
+            abort(400, f"File '{item.path}' contains invalid characters.")
+        if not is_supported_extension(item.path):
+            abort(400, f"Unsupported extension of '{item.path}' file.")
```

Comment:
I'd suggest
```suggestion
            abort(400, f"Unsupported file type detected: {item.path}")
```
`File type` is easier to understand than `extension`. While it is obvious to us, end users might not know what that refers to :)

Reply:
Shall we add also a hint what to do to unblock upload? `Please remove the file from the upload folder or try zipping the file before uploading.`
""

Reply:
Yeah, good idea! 👍🏻 How about `Please remove the file or try compressing it into a ZIP file before uploading.`?